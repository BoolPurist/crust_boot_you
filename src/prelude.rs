use std::io;

pub use crate::constants;
pub use anyhow::Context;

pub type IoResult<T = ()> = Result<T, io::Error>;
pub type AppError = anyhow::Error;
pub type AppResult<T = ()> = Result<T, AppError>;

pub type ReturnToUser = AppResult<String>;
pub type PathResult = AppResult<PathBuf>;
pub type AppIoResult<T = ()> = Result<T, AppIoError>;

pub use crate::app_traits::file_manipulator::FileManipulator;
pub use crate::app_traits::path_provider::PathProvider;
pub use crate::file_management::AppIoError;
pub use crate::NotEmptyText;
pub use std::path::{Path, PathBuf};

#[cfg(test)]
pub use mockall::automock;
