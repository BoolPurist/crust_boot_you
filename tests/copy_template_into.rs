mod common;
use common::prelude::*;
use crust_boot_you::{
    cli::{InitKind, LoadTemplateArg},
    handle_commands,
    template_augmentation::{console_fetcher::TestConsoleFetcher, RegexTemplateAugmentor},
};
use map_macro::hash_map;

#[test]
#[named]
fn copy_at_target() {
    let setup = TestSetupBuilder::new(actual_expected!())
        .suffix_cwd(PathBuf::from("cwd").join("some_where"))
        .build();

    let to_copy_from = ValidTemplateName::new("to_copy".to_string()).unwrap();
    let init_kind = InitKind::OnlyEmpty;
    let arg = LoadTemplateArg::new(to_copy_from, init_kind);
    let mut store: RegexTemplateAugmentor<TestConsoleFetcher> =
        RegexTemplateAugmentor::from_fake(Default::default());
    let output = handle_commands::handle_load_template(
        setup.path_provider(),
        setup.os_mani(),
        &mut store,
        &arg,
    )
    .unwrap();
    setup.assert_with_expected();

    insta_display_filter_random_tmp!(output);
}
#[test]
#[named]
fn copy_err_cwd_not_empty() {
    let setup = TestSetupBuilder::new(actual_expected!())
        .suffix_cwd(PathBuf::from("cwd").join("not_empty"))
        .build();

    let to_copy_from = ValidTemplateName::new("a".to_string()).unwrap();
    let init_kind = InitKind::OnlyEmpty;
    let arg = LoadTemplateArg::new(to_copy_from, init_kind);
    let mut store: RegexTemplateAugmentor<TestConsoleFetcher> =
        RegexTemplateAugmentor::from_fake(Default::default());
    let output = handle_commands::handle_load_template(
        setup.path_provider(),
        setup.os_mani(),
        &mut store,
        &arg,
    )
    .unwrap_err();

    setup.assert_with_expected();
    insta_display_filter_random_tmp!(output);
}

#[test]
#[named]
fn copy_err_template_not_found() {
    let setup = TestSetupBuilder::new(actual_expected!())
        .suffix_cwd(PathBuf::from("cwd").join("not_empty"))
        .build();

    let to_copy_from = ValidTemplateName::new("not_there".to_string()).unwrap();
    let init_kind = InitKind::OnlyEmpty;
    let arg = LoadTemplateArg::new(to_copy_from, init_kind);
    let mut store: RegexTemplateAugmentor<TestConsoleFetcher> =
        RegexTemplateAugmentor::from_fake(Default::default());
    let output = handle_commands::handle_load_template(
        setup.path_provider(),
        setup.os_mani(),
        &mut store,
        &arg,
    )
    .unwrap_err();

    setup.assert_with_expected();
    insta_display_filter_random_tmp!(output);
}

#[test]
#[named]
fn initializ_with_no_name_conflicts() {
    let setup = TestSetupBuilder::new(actual_expected!())
        .suffix_cwd(PathBuf::from("cwd").join("without_name_conflicts"))
        .build();

    let to_copy_from = ValidTemplateName::new("with_placeholders".to_string()).unwrap();
    let init_kind = InitKind::NoNameConflicts;
    let arg = LoadTemplateArg::new(to_copy_from, init_kind);
    let values = hash_map! {"user_name".to_string() => "pattern".to_string()};

    let mut store: RegexTemplateAugmentor<TestConsoleFetcher> =
        RegexTemplateAugmentor::from_fake(values);
    let output = handle_commands::handle_load_template(
        setup.path_provider(),
        setup.os_mani(),
        &mut store,
        &arg,
    )
    .unwrap();

    setup.assert_with_expected();
    insta_display_filter_random_tmp!(output);
}
