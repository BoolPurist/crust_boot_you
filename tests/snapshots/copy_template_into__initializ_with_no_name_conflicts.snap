---
source: tests/copy_template_into.rs
expression: output
---
Folder "/[TMP_ROOT]/cwd/without_name_conflicts" filled with content from Template (with_placeholders)
